controllers/csm_controller.go

```diff
@@ -620,6 +643,19 @@ func (r *ContainerStorageModuleReconciler) SyncCSM(ctx context.Context, cr csmv1
 			}
 		}
 
+		// if Observability is enabled, create or update obs components: topology, metrics of PowerScale and PowerFlex
```

Comment:
for this block codes, how about Extract them into a function

Reply:
because in the future, metrics codes will be added to this block too

Reply:
addressed